corehq/apps/ip_access/models.py

```diff
@@ -0,0 +1,46 @@
+import geoip2.webservice
+
+from django.conf import settings
+from django.contrib.postgres.fields import ArrayField
+from django.db import models
+
+from corehq.util.metrics import metrics_counter
+
+
+class IPAccessConfig(models.Model):
+    domain = models.CharField(max_length=126, db_index=True, unique=True)
+    country_allowlist = ArrayField(models.CharField(max_length=2), default=list)
+    ip_allowlist = ArrayField(models.GenericIPAddressField(), default=list)
+    ip_denylist = ArrayField(models.GenericIPAddressField(), default=list)
+    comment = models.TextField(blank=True)
+    created_on = models.DateTimeField(auto_now_add=True)
+    updated_on = models.DateTimeField(auto_now=True)
+
+    def is_allowed(self, ip_address):
+        should_check_country = True
+        if not self.country_allowlist:
```

Comment:
Don't we only need to check Maxmind license only if `country_allowlist` is populated? 

Should the logic be (I'm assuming an empty self.country_allowlist means allow no countries)


```
def is_allowed(self, ip_address):
    if ip_address in self.ip_denylist:
        return False

    if ip_address in self.ip_allowlist:
        return True

    if self.country_allowlist and not settings.MAXMIND_LICENSE_KEY:
        return False

    return is_in_country(ip_address, self.country_allowlist)

```

Reply:
Yeah doesn't this change it so `country_allowlist = []` means no countries are allowed?